src/application/predictor.hpp

```diff
@@ -104,12 +108,45 @@ class Predictor {
       Log::Fatal("Could not recognize the data format of data file %s", data_filename);
     }
 
+    std::unordered_map<int, int> feature_names_map_;
+    if(has_header) {
+      std::ifstream tmp_file;
+      tmp_file.open(data_filename);
+      if (!tmp_file.is_open()) {
+        Log::Fatal("Data file %s doesn't exist'", data_filename);
+      }
+      std::string line1;
+      if (!tmp_file.eof()) {
+        std::getline(tmp_file, line1);
+        std::vector<std::string> header = Common::Split(line1.c_str(), "\t");
+        header.erase(header.begin() + boosting_->LabelIdx());
+        for(int i = 0; i < static_cast<int>(header.size()); ++i) {
+          for(int j = 0; j < static_cast<int>(boosting_->FeatureNames().size()); ++j) {
+            if(header[i] == boosting_->FeatureNames()[j]) {
+              feature_names_map_[i] = j;
+              break;
+            }
+          }
+        }
+      }
+    }
```

Comment:
these is too heavy, refer to : https://github.com/Microsoft/LightGBM/blob/master/src/io/dataset_loader.cpp#L29-L32